apps/web/src/pages/templates/filter/FilterModal.tsx

```diff
@@ -25,163 +25,165 @@ export function FilterModal({
-                <Controller
-                  control={control}
-                  name={`steps.${stepIndex}.filters.0.children.${index}.operator`}
-                  render={({ field }) => {
-                    return (
-                      <Select
-                        placeholder="Operator"
-                        data={[
-                          { value: 'EQUAL', label: 'Equal' },
-                          { value: 'NOT_EQUAL', label: 'Not equal' },
-                          { value: 'LARGER', label: 'Larger' },
-                          { value: 'SMALLER', label: 'Smaller' },
-                          { value: 'LARGER_EQUAL', label: 'Larger or equal' },
-                          { value: 'SMALLER_EQUAL', label: 'Smaller or equal' },
-                          { value: 'IN', label: 'Contains' },
-                          { value: 'NOT_IN', label: 'Not contains' },
-                        ]}
-                        {...field}
-                      />
-                    );
-                  }}
-                />
-              </Grid.Col>
-              <Grid.Col span={2}>
-                <Controller
-                  control={control}
-                  name={`steps.${stepIndex}.filters.0.children.${index}.value`}
-                  render={({ field, fieldState }) => {
-                    return <Input {...field} error={fieldState.error?.message} placeholder="Value" />;
-                  }}
-                />
-              </Grid.Col>
-              <Grid.Col span={1}>
-                <DeleteStepButton
-                  variant="outline"
-                  size="md"
-                  mt={30}
-                  onClick={() => {
-                    remove(index);
-                  }}
-                >
-                  <Trash />
-                </DeleteStepButton>
-              </Grid.Col>
-            </Grid>
-          );
-        })}
-        <div>
-          <Group position="right">
-            <Button variant="outline" size="md" mt={30} onClick={() => cancel()}>
-              Cancel
-            </Button>
-            <Button mt={30} size="md" onClick={() => confirm()}>
-              Add
-            </Button>
-          </Group>
-        </div>
-      </Modal>
-    </>
+              );
+            }}
+          />
+        </Grid.Col>
+        <Grid.Col span={7} />
+        <Grid.Col span={2}>
+          <FilterButton
+            variant="outline"
+            size="md"
+            mt={30}
+            onClick={() => {
+              append({
+                operator: 'EQUAL',
+                on: 'payload',
+              });
+            }}
+          >
+            Create rule
+          </FilterButton>
+        </Grid.Col>
+      </Grid>
+      <Divider
+        sx={{
+          marginTop: '15px',
+          marginBottom: '15px',
+        }}
+      />
+      {fields.map((item, index) => {
+        return (
+          <Grid columns={10} key={`filter-${index}`} align="center" gutter="xs">
```

Comment:
fields is an array with objects that have an `id` field, please use it instead like `key={item.id}`

Reply:
Fixed